### PR TITLE
Allow flags to be specified for transformers and make labels lowercase

### PR DESCRIPTION
**Pull request by invakid404**
This PR adds a way to specify flags to transformers via the configuration and treats labels as case-insensitive.

### CODE REVIEW

File: src/releaseNotes.ts [hunk 1]
```diff
@@ -176,7 +176,7 @@ export class ReleaseNotes {
         mergeCommitSha: '',
         author: commit.author || '',
         repoName: '',
-        labels: [],
+        labels: new Set(),
```
Comment (mikepenz) [inline src/releaseNotes.ts:179]: @invakid404 could you please elaborate why the change to `Set` would be needed?

File: src/pullRequests.ts [hunk 2]
```diff
@@ -159,3 +141,31 @@ export function sortPullRequests(
   }
   return pullRequests
 }
+
+const mapPullRequest = (
+  pr: PullData | Unpacked<PullsListData>
+): PullRequestInfo => ({
```
Comment (mikepenz) [inline src/pullRequests.ts:147]: I believe this (`Unpacked`) is not required to allow the user to define the flags for the transformers / have labels be lowercase. 
Maybe I am missing something, could you please describe the main benefit?

File: src/releaseNotes.ts (hunk 1 above)
Comment (invakid404) [reply src/releaseNotes.ts:179]: It's nothing too important, but it's generally better to store labels in a set to ensure we don't store duplicates and speed up lookup time when categorizing the PRs.

Comment (mikepenz) [reply src/releaseNotes.ts:179]: Fair point. Agree, `Set` is better in this case then :)

File: src/pullRequests.ts (hunk 2 above)
Comment (invakid404) [reply src/pullRequests.ts:147]: This is just a minor refactor I did while making the change to store labels in a set. Since there was a duplication of all of this mapping logic, I thought it's reasonable to move it to a function. The Unpacked<T> type is simply used to get the type of a single entry of the return type of the list endpoint, since they differ for some reason.

Comment (mikepenz) [reply src/pullRequests.ts:147]: Maybe I overlook it, but I don't see `mapPullRequest` being used twice. 

Could you maybe exclude this change from the PR, not sure yet if we need the `Unpacked` type if we keep the old definition for this here.

**Comment by mikepenz** [comment]
@invakid404 could you please rebase your PR on top of dev. as there were some updates required for the tests

**Comment by invakid404** [comment]
> @invakid404 could you please rebase your PR on top of dev. as there were some updates required for the tests

Sure thing, I'll deal with it in ~30 minutes when I'm back in front of the computer.

**Comment by invakid404** [comment]
@mikepenz This branch is already up-to-date with develop. Only thing I had forgotten to push was the updated dist files. Can you re-run the tests to see if it passes?

File: src/pullRequests.ts (hunk 2 above)
Comment (invakid404) [reply src/pullRequests.ts:147]: Oh, my bad, I've forgotten to use it in the other place.

**Comment by invakid404** [comment]
> @mikepenz This branch is already up-to-date with develop. Only thing I had forgotten to push was the updated dist files. Can you re-run the tests to see if it passes?

Looks like it still fails... Do you have a clue why?

**Comment by mikepenz** [comment]
@invakid404 I believe it's because labels are now non case sensitive retrieved from github, but the configuration itself still handles them case sensitive. 

will fix this.

**Comment by mikepenz** [comment]
Thank you so much for the PR!

Had done 2 more commits to it:
https://github.com/mikepenz/release-changelog-builder-action/commit/4d47219031bcb044bfed3aeb30906f8dba3d14aa
https://github.com/mikepenz/release-changelog-builder-action/commit/99f650ca022fbb55fe73de9f56a0ee892d438c7e
